ui/packages/catalog-ui-search/src/main/webapp/js/CQLUtils.js

```diff
@@ -9,254 +9,295 @@
  * <http://www.gnu.org/licenses/lgpl.html>.
  *
  **/
-/*global define*/
 /*jshint bitwise: false*/
-define([
-    'jquery',
-    'js/cql',
-    'js/DistanceUtils.js',
-    'component/singletons/metacard-definitions'
-], function ($, cql, DistanceUtils, metacardDefinitions) {
+const $ = require('jquery');
+const cql = require('js/cql');
+const DistanceUtils = require('js/DistanceUtils.js')
 
-    return {
-        sanitizeForCql: function (text) {
-            return text.split('[').join('(').split(']').join(')').split("'").join('').split('"').join('');
-        },
+function sanitizeForCql(text) {
+    return text.split('[').join('(').split(']').join(')').split("'").join('').split('"').join('');
+}
 
-        sanitizeGeometryCql: function (cqlString) {
-            //sanitize polygons
-            let polygons = cqlString.match(/'POLYGON\(\((-?[0-9]*.?[0-9]* -?[0-9]*.?[0-9]*,?)*\)\)'/g);
-            if (polygons) {
-                polygons.forEach((polygon) => {
-                    cqlString = cqlString.replace(polygon, polygon.replace(/'/g, ''));
-                });
-            }
+function generateAnyGeoFilter(property, model) {
+    switch (model.type) {
+        case 'LINE':
+            return {
+                type: 'DWITHIN',
+                property: property,
+                value: 'LINESTRING' +
+                sanitizeForCql(JSON.stringify(lineToCQLLIne(model.line))),
+                distance: DistanceUtils.getDistanceInMeters(model.lineWidth, model.lineUnits)
+            };
+        case 'POLYGON':
+            return {
+                type: 'INTERSECTS',
+                property: property,
+                value: 'POLYGON(' +
+                sanitizeForCql(JSON.stringify(polygonToCQLPolygon(model.polygon))) + ')'
+            };
+        case 'MULTIPOLYGON':
+            var poly = 'MULTIPOLYGON' +
```

Comment:
Does this not get wrapped in `(` and `)` like the others?

Reply:
No, the logic in `polygonToCQLMultiPolygon` was modified to return a triply nested array. I'll modify the `POLYGON` case though (and the `polygonToCQLPolygon` logic) to make these more consistent with each other and more readable.